src/github.com/matrix-org/dendrite/roomserver/storage/storage.go

```diff
@@ -35,3 +36,71 @@ func (d *Database) PartitionOffsets(topic string) ([]types.PartitionOffset, erro
 func (d *Database) SetPartitionOffset(topic string, partition int32, offset int64) error {
 	return d.statements.upsertPartitionOffset(topic, partition, offset)
 }
+
+// StoreEvent implements input.EventDatabase
+func (d *Database) StoreEvent(event gomatrixserverlib.Event) error {
+	var (
+		roomNID          int64
+		eventTypeNID     int64
+		eventStateKeyNID int64
+		eventNID         int64
+		err              error
+	)
+
+	if roomNID, err = d.assignRoomNID(event.RoomID()); err != nil {
+		return err
+	}
+
+	if eventTypeNID, err = d.assignEventTypeNID(event.Type()); err != nil {
+		return err
+	}
+
+	eventStateKey := event.StateKey()
+	if eventStateKey != nil {
+		if eventStateKeyNID, err = d.assignStateKeyNID(*eventStateKey); err != nil {
+			return err
+		}
+	}
+
+	eventID := event.EventID()
+	referenceSHA256 := event.EventReference().EventSHA256
+
+	if eventNID, err = d.statements.insertEvent(roomNID, eventTypeNID, eventStateKeyNID, eventID, referenceSHA256); err != nil {
+		return err
+	}
+
+	return d.statements.insertEventJSON(eventNID, event.JSON())
```

Comment:
Why do we have 2 functions: one for `insertEvent` and one for `insertEventJSON`? Why not just use 1?